test/controllers/MovementsControllerSpec.scala

```diff
@@ -799,8 +885,99 @@ class MovementsControllerSpec extends SpecBase with ScalaCheckPropertyChecks wit
+          )
+          .build()
+
+        running(application) {
+
+          val dateOfPrep = LocalDate.now()
+          val timeOfPrep = LocalTime.of(1, 1)
+
+          val requestXmlBody =
+            <CC007A>
+              <DatOfPreMES9>{Format.dateFormatted(dateOfPrep)}</DatOfPreMES9>
+              <TimOfPreMES10>{Format.timeFormatted(timeOfPrep)}</TimOfPreMES10>
+              <SynVerNumMES2>1</SynVerNumMES2>
+              <HEAHEA>
+                <DocNumHEA5>MRN</DocNumHEA5>
+              </HEAHEA>
+            </CC007A>
+
+          val request = FakeRequest(PUT, routes.MovementsController.putArrival(initializedArrival.arrivalId).url).withXmlBody(requestXmlBody.map(trim))
+
+          val result = route(application, request).value
+
+          status(result) mustEqual BAD_REQUEST
+
+        }
+      }
+
+      "must return InternalServerError if the message has been rejected from EIS due to virus found or invalid token" in {
+        val mockArrivalMovementRepository = mock[ArrivalMovementRepository]
+        val mockMessageConnector          = mock[MessageConnector]
+        val mockLockRepository            = mock[LockRepository]
+        val mockSubmitMessageService      = mock[SubmitMessageService]
+
+        when(mockLockRepository.lock(any())).thenReturn(Future.successful(true))
+        when(mockLockRepository.unlock(any())).thenReturn(Future.successful(()))
+        when(mockArrivalMovementRepository.get(any())).thenReturn(Future.successful(Some(initializedArrival)))
+
+        when(mockSubmitMessageService.submitIe007Message(any(), any(), any(), any())(any()))
+          .thenReturn(Future.successful(SubmissionProcessingResult.SubmissionFailureRejected(VirusFoundOrInvalidToken)))
+
+        val application = baseApplicationBuilder
+          .overrides(
+            bind[LockRepository].toInstance(mockLockRepository),
+            bind[ArrivalMovementRepository].toInstance(mockArrivalMovementRepository),
+            bind[MessageConnector].toInstance(mockMessageConnector),
+            bind[SubmitMessageService].toInstance(mockSubmitMessageService)
+          )
+          .build()
+
+        running(application) {
+
+          val dateOfPrep = LocalDate.now()
+          val timeOfPrep = LocalTime.of(1, 1)
+
+          val requestXmlBody =
+            <CC007A>
+              <DatOfPreMES9>{Format.dateFormatted(dateOfPrep)}</DatOfPreMES9>
+              <TimOfPreMES10>{Format.timeFormatted(timeOfPrep)}</TimOfPreMES10>
+              <SynVerNumMES2>1</SynVerNumMES2>
+              <HEAHEA>
+                <DocNumHEA5>MRN</DocNumHEA5>
+              </HEAHEA>
+            </CC007A>
+
+          val request = FakeRequest(PUT, routes.MovementsController.putArrival(initializedArrival.arrivalId).url).withXmlBody(requestXmlBody.map(trim))
+
+          val result = route(application, request).value
+
+          status(result) mustEqual INTERNAL_SERVER_ERROR
+
+        }
+      }
```

Comment:
Why is this an internal server error? This service did not create this payload but is passing along the payload from upstream

Reply:
The controller recodes the status code to 500 before sending upstream.

Reply:
After discussion we think that the spec is not granular enough and downstream from the router should instead return a 401 if the auth tokens are missing/invalid

If we cannot resolve this will be good to go

Reply:
Looking over the spec this is a misunderstanding on my part, this is good to the spec